index.bs

```diff
@@ -795,6 +741,528 @@ the following list:
+
+  <dl class="switch">
+   : |type| indicates that the data is Text
+   :: a new {{DOMString}} containing |data|
+
+   : |type| indicates that the data is Binary
+   :: a new {{Uint8Array}} object, created in the [=relevant Realm=] of the {{WebSocketStream}}
+      object, whose contents are |data|
+  </dl>
+
+ 1. [=ReadableStream/Enqueue=] |chunk| into |this|'s [=WebSocketStream/readable stream=].
+ 1. Apply backpressure to the WebSocket.
+
+ Note: Applying backpressure <span class=allow-2119>should</span> result in no new WebSocket
+ messages being handled until the backpressure is released. After an implementation-defined amount
+ of data is buffered, the client will stop reading from the TCP/IP socket resulting in network-layer
+ backpressure being applied.
+ </div>
+
+<hr>
+
+When [=the WebSocket closing handshake is started=], the user agent must [=queue a task=] to change
+the [=WebSocketStream/ready state=] to {{WebSocket/CLOSING}} if it is not already. [[!WSP]]
+
+<hr>
+
+<div algorithm="handle WebSocketStream connection close">
+
+When [=the WebSocket connection is closed=] for |this|, possibly [=cleanly=], the user agent must
+ [=queue a task=] to run the following substeps:
+
+ 1. Change the [=WebSocketStream/ready state=] to {{WebSocket/CLOSED}} (3).
+ 1. If |this|'s [=WebSocketStream/was ever connected flag=] is not set, then [=reject=] |this|'s
+    [=opened promise=] with a [=new=] WebSocketError.
+ 1. Let |code| be [=the WebSocket connection close code=].
+ 1. Let |reason| be the result of applying [=UTF-8 decode without BOM=] to [=the WebSocket
+    connection close reason=].
+ 1. If the connection was closed [=cleanly=],
+  1. [=ReadableStream/Close=] |this|'s [=WebSocketStream/readable stream=].
+  1. [=WritableStream/Error=] |this|'s [=WebSocketStream/writable stream=] with an
+     "{{InvalidStateError}}" {{DOMException}} indicating that a closed {{WebSocketStream}} cannot be
+     written to.
+  1. [=Resolve=] |this|'s [=closed promise=] with a [=new=] {{WebSocketCloseInfo}} constructed with
+     |code| and |reason|.
+ 1. Otherwise,
+  1. [=ReadableStream/Error=] |this|'s [=WebSocketStream/readable stream=] with a [=new=]
+     {{WebSocketError}} constructed with |code| and |reason|.
+  1. [=WritableStream/Error=] |this|'s [=WebSocketStream/writable stream=] with a [=new=]
+     {{WebSocketError}} constructed with |code| and |reason|.
+  1. [=Reject=] |this|'s [=closed promise=] with a [=new=] {{WebSocketError}} constructed with
+     |code| and |reason|.
+</div>
+
+<hr>
+
+The [=task source=] for all [=tasks=] <a lt="queue a task">queued</a> in this section is the
+[=WebSocket task source=]
+
+<hr>
+
+
+# Stream operations # {#stream-operations}
+
+<div algorithm="pull algorithm">
+
+To <dfn>pull bytes</dfn> from a WebSocket, if the WebSocket is currently applying backpressure,
+release backpressure.
+
+ Note: If any messages are queued, one will be handled immediately as a result.
```

Comment:
Wondering if we should just use `<p class="note">` here and in other places to be consistent.

Reply:
I would rather change it to "Note:" everywhere as I prefer the simpler syntax. Would you mind if I did that in a follow-up PR?

Reply:
The thing I dislike about "Note:" is that it introduces a duplicate "Note:" in the actual text, in addition to the dark-green "Note" in the corner.

Reply:
Oh, I failed to notice that. That's not good. I changed them all to `<p class=note>`.